assets/src/components/parts/janus-hub-spoke/HubSpoke.tsx

```diff
@@ -0,0 +1,378 @@
+/* eslint-disable react/prop-types */
+import React, { CSSProperties, useCallback, useEffect, useState } from 'react';
+import { CapiVariableTypes } from '../../../adaptivity/capi';
+import {
+  NotificationType,
+  subscribeToNotification,
+} from '../../../apps/delivery/components/NotificationContext';
+import { contexts } from '../../../types/applicationContext';
+import { PartComponentProps } from '../types/parts';
+import { Item, JanusHubSpokeItemProperties, hubSpokeModel } from './schema';
+
+const SpokeItemContentComponent: React.FC<any> = ({ spokeLabel }) => {
+  return spokeLabel;
+};
+
+const SpokeItemContent = React.memo(SpokeItemContentComponent);
+
+export const SpokeItems: React.FC<JanusHubSpokeItemProperties> = ({
+  spokeLabel,
+  state,
+  itemId,
+  layoutType,
+  totalItems,
+  idx,
+  overrideHeight,
+  columns = 1,
+  index,
+  verticalGap = 0,
+  onSelected,
+  val,
+  IsCompleted,
+}) => {
+  const spokeItemStyles: CSSProperties = {};
+  if (layoutType === 'horizontalLayout') {
+    if (columns === 1) {
+      spokeItemStyles.width = `calc(${Math.floor(100 / totalItems)}% - 6px)`;
+    } else {
+      spokeItemStyles.width = `calc(100% / ${columns} - 6px)`;
+    }
+    if (idx !== 0) {
+      spokeItemStyles.left = `calc(${(100 / totalItems) * idx}% - 6px)`;
+    }
+    spokeItemStyles.position = `absolute`;
+
+    spokeItemStyles.display = `inline-block`;
+  }
+  if (layoutType === 'verticalLayout' && overrideHeight) {
+    spokeItemStyles.height = `calc(${100 / totalItems}%)`;
+  }
+  if (layoutType === 'verticalLayout' && verticalGap && index > 0) {
+    spokeItemStyles.marginTop = `${verticalGap}px`;
+  }
+  return (
+    <React.Fragment>
+      <div style={spokeItemStyles} className={` hub-spoke-item`}>
+        <button
+          type="button"
+          style={{ width: '100%' }}
+          onClick={() => {
+            if (onSelected) onSelected(val);
+          }}
+          className="btn btn-primary"
+        >
+          {IsCompleted && (
+            <span style={{ float: 'left', paddingLeft: '10px' }} className={'fa fa-check-circle'}>
+              &nbsp;
+            </span>
+          )}
+          <SpokeItemContent itemId={itemId} spokeLabel={spokeLabel} state={state} />
+        </button>
+      </div>
+      {layoutType !== 'horizontalLayout' && <br style={{ padding: '0px' }} />}
+    </React.Fragment>
+  );
+};
+
+interface SpokeOptionModel extends Item {
+  index: number;
+  value: number;
+}
+
+const HubSpoke: React.FC<PartComponentProps<hubSpokeModel>> = (props) => {
+  const [model, setModel] = useState<any>(Array.isArray(props.model) ? props.model : {});
```

Comment:
There are a lot of different `useState` defined here, maybe it makes sense to encapsulate all of this state mutation logic in a reducer and `useReducer` instead?